app/models/katello/alternate_content_source.rb

```diff
@@ -165,5 +163,13 @@ def validate_ssl_ids
         end
       end
     end
+
+    # Check that products are not present for custom or rhui ACS's
+    # Requires complex/custom error message
```

Comment:
Comments not necessary..`errors.add(:product_ids, "cannot be set for custom or rhui ACS")` is good enough explanation for readability..